composer.json

```diff
@@ -1,55 +1,71 @@
-		]
-	},
-	"require": {
-		"phpoffice/phpspreadsheet": "^5.1",
-		"ext-json": "*"
-	}
+  "name": "nextcloud/tables",
+  "description": "This app is for managing data in tables.",
+  "type": "project",
+  "license": "AGPL",
+  "authors": [
+    {
+      "name": "Florian Steffens",
+      "email": "florian@nextcloud.com"
+    }
+  ],
+  "autoload": {
+    "psr-4": {
+      "OCA\\Tables\\": "lib/"
+    }
+  },
+  "require-dev": {
+    "nextcloud/coding-standard": "^v1.4.0",
+    "nextcloud/ocp": "dev-stable30",
+    "staabm/annotate-pull-request-from-checkstyle": "^1.8.6",
+    "phpunit/phpunit": "^9",
+    "psalm/phar": "^5.26.1"
+  },
+  "config": {
+    "optimize-autoloader": true,
+    "classmap-authoritative": true,
+    "platform": {
+      "php": "8.1"
+    },
+    "allow-plugins": {
+      "bamarni/composer-bin-plugin": true
+    }
+  },
+  "scripts": {
+    "test": [
+      "@test:unit"
+    ],
+    "test:unit": "./vendor/bin/phpunit -c tests/unit/phpunit.xml",
+    "test:unit:local": "TEST_MODE=local composer test:unit",
+    "lint": "find . -name \\*.php -not -path './vendor/*' -not -path './build/*' -print0 | xargs -0 -n1 php -l",
+    "cs:check": "php-cs-fixer fix --dry-run --diff",
+    "cs:fix": "php-cs-fixer fix",
+    "psalm": "./vendor/bin/psalm.phar --show-info=false --no-cache",
+    "psalm:update-baseline": "./vendor/bin/psalm.phar --update-baseline",
+    "psalm:fix": "./vendor/bin/psalm.phar --no-cache --alter --issues=InvalidReturnType,InvalidNullableReturnType,MismatchingDocblockParamType,MismatchingDocblockReturnType,MissingParamType,InvalidFalsableReturnType",
+    "psalm:fix:dry": "./vendor/bin/psalm.phar --no-cache --alter --issues=InvalidReturnType,InvalidNullableReturnType,MismatchingDocblockParamType,MismatchingDocblockReturnType,MissingParamType,InvalidFalsableReturnType --dry-run",
+    "openapi": "generate-spec --verbose && (npm run typescript:generate || echo 'Please manually regenerate the typescript OpenAPI models')",
+    "scoper:update-deps": "./update-scoper-dependencies.sh",
+    "post-install-cmd": [
+      "composer bin all install --ansi",
+      "vendor/bin/php-scoper add-prefix --force",
+      "@php .lib-vendor-organizer.php lib/Vendor/ OCA\\\\Tables\\\\Vendor",
+      "cat .scoper-production-dependencies | uniq | xargs -I {} rm -Rf vendor/{} # Remove origins",
+      "cat .scoper-production-dependencies | cut -d / -f1 | uniq | xargs -I {} rmdir vendor/{} || true # Remove empty directories",
+      "composer dump-autoload -o"
+    ],
+    "post-update-cmd": [
+      "composer bin all update --ansi",
+      "vendor/bin/php-scoper add-prefix --force",
+      "@php .lib-vendor-organizer.php lib/Vendor/ OCA\\\\Tables\\\\Vendor",
+      "cat .scoper-production-dependencies | uniq | xargs -I {} rm -Rf vendor/{} # Remove origins",
+      "cat .scoper-production-dependencies | cut -d / -f1 | uniq | xargs -I {} rmdir vendor/{} || true # Remove empty directories",
+      "composer dump-autoload -o"
+    ]
```

Comment:
At what point does one run these new commands? After installing any new library?

Reply:
Oooh, cool. What about for `scoper:update-deps`? When do I run it?

Reply:
After installing, updating or removing a dependency. But CI will also tell you.